ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -979,6 +980,18 @@ export default class ConfirmTransactionBase extends Component {
     window.removeEventListener('beforeunload', this._beforeUnloadForGasPolling);
   };
 
+  UNSAFE_componentWillMount() {
+    const { txData: { chainId: txChainId } = {}, chainId } = this.props;
+
+    // If the user somehow finds themselves seeing a confirmation
+    // on a network which is not presently selected, throw
+    if (txChainId === undefined || txChainId !== chainId) {
+      throw new Error(
+        `Confirmation displaying on wrong chain.  Expected ${txChainId}, current chain is ${chainId}.`,
+      );
+    }
+  }
+
```

Comment:
why in `UNSAFE_componentWillMount` instead of `componentDidMount`? Is the idea that there is some small risk that the confirmation renders and user can confirm before that `componentDidMount` hook happens?

Reply:
My hope was to throw this error ASAP; there's point in waiting to mount if we know we're going to throw anyway.  The result in either case will be the error stack trace.

Reply:
I guess my concern is why introduce a lifecyclehook that the react team has explicitly say may be buggy in future versions when componentDidMount will effectively serve the same purpose?